app/assets/javascripts/angular/controllers/pages/recipe_finder_controller.js.erb

```diff
@@ -17,11 +19,167 @@ app.controller('RecipeFinderCtrl', ['$scope', '$http', '$routeParams', function(
+
+    $scope.courses = ["Appetizers","Mains","Desserts","Drinks","Sides","Soups","Salads","Bread"];
+
+    $scope.courses_hidden = ["Dips and Spreads", "Snack", "Wraps and Rolls Appetizers", "Antipasto", "Bread and Pastry Appetizers", "Crackers", "Deviled Eggs", "Pasta Appetizers", "Pastry Appetizers", "Dill Pickles and Sweet Pickles", "Pickled Eggs", "Popcorn", "Slow Cooker Appetizers", "Snacks for Kids", "Spicy Appetizers", "Bean and Pea Appetizers", "Cheese Appetizers", "Fruit Appetizers", "Meat and Poultry Appetizers", "Mushroom Appetizers", "Nuts and Seeds Appetizers", "Olive Appetizers", "Seafood Appetizers", "Vegetable Appetizers","Yeast Bread", "Bread Machine", "Holiday Bread", "Pastries", "Pizza Dough and Crusts", "Quick Bread","Deep Fried Main", "Eggplant Parmesan", "Steaks and Chops", "Stir-Fry", "Stuffed Main", "Burgers", "Calzones", "Casseroles", "Roasts", "Curry Main", "Meatballs", "Meatloaf", "Pizza", "Quiche", "Savory Pies", "Seafood Main", "Main Dish Soups and Stews", "Ribs", "Sandwiches and Wraps", "Main Dish Salads", "Pasta Main","Candy", "Cobbler", "Mousse", "Pie", "Crisps and Crumbles", "Custards and Puddings", "Cake", "Cookies", "Chocolate Desserts", "Frozen Desserts", "Fruit Desserts", "Nut Desserts", "Specialty Desserts", "Frostings and Fillings","Smoothies", "Coffee Drinks", "Eggnog", "Hot Chocolate", "Juice", "Lemonade", "Liqueurs", "Punch", "Shakes and Floats", "Shots", "Mocktails", "Mulled Wine", "Sangria", "Slushies", "Tea Drinks", "Beer Drinks", "Cider", "Cocktails","Beef and Pork Salads", "Coleslaw", "Curry Salad", "Egg Salad", "Fruit Salads", "Green Salads", "Grain Salads", "Pasta Salad", "Potato Salad", "Seafood Salads", "Taco Salad", "Vegetable Salads", "Bean and Pea Salads", "Waldorf Salad","Potato Side", "Applesauce", "Bean and Pea Side", "Curry Side", "Dumplings", "Fries", "Grain Side", "Rice Side", "Sauces and Condiments", "Side Dish Casseroles", "Stuffing and Dressing", "Vegetable Side", "Grits", "Polenta","Bisque", "Broth and Stock", "Chili", "Chowder", "Soup", "Stew"].sort();
+
+    $scope.occasions = ["Events and Gatherings", "BBQ and Grilling", "Everyday"];
+
+    $scope.occasions_hidden = ["Christmas", "Thanksgiving", "Easter", "Super Bowl®", "Labor Day", "Valentine's Day", "4th of July", "New Year's", "Halloween", "Mother's Day", "Mardi Gras", "Memorial Day", "Father's Day", "Hanukkah", "St. Patrick's Day", "Cinco de Mayo", "Passover", "Rosh Hashanah", "Ramadan", "Purim", "Lunar New Year", "Kwanzaa", "Veteran's Day"];
+  }
+
+    // keyword search
+  { 
+    $scope.search_parameters.keywords = [];
+    $scope.moreKeyword = function() {
```

Comment:
It would look cleaner to use verbs for functions (=actions) and nouns for items: e.g. ```$scope.addKeyword = function ()``` where ```$scope.newKeyword``` is your new keyword